src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> tail;
+    private Node<T> head;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value);
+        if (head == null) {
+            tail = newNode;
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        size++;
+        checkIndex(index);
+        size--;
+        Node<T> nodeForeAdd = new Node<>(value);
+        if (head == null) {
+            head = tail = nodeForeAdd;
+        } else if (index == 0) {
+            nodeForeAdd.next = head;
+            head = nodeForeAdd;
+        } else if (index == size) {
+            nodeForeAdd.prev = tail;
+            tail.next = nodeForeAdd;
+            tail = nodeForeAdd;
+        } else {
+            insert(value,index);
+        }
+        size++;
```

Comment:
Please try to simplify this logic. Also consider reusing your other implemented methods (for example, your add() with 1 parameter, getByIndex() ...)